File: selenium-webdriver-nodejs-mocha/spec/03_passenger_spec.js
var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;
var test = require('selenium-webdriver/testing');
var assert = require('assert');

var driver;
const timeOut = 15000;

String.prototype.contains = function(it) {
  return this.indexOf(it) != -1;
};

var helper = require('../test_helper');

// An example in below:  var FlightPage = require('../pages/flight_page.js')
// BEGIN: import pages

var FlightPage = require('../pages/flight_page.js')
var PassengerPage = require('../pages/passenger_page.js')

// END: import pages


describe('Passenger', function() {

  before(async function() {
    this.timeout(timeOut);
    driver = new webdriver.Builder().forBrowser(helper.browserType()).setChromeOptions(helper.chromeOptions()).build();
    driver.manage().window().setRect({width: 1027, height: 700, x: 30, y: 78})
  });

  beforeEach(async function() {
    this.timeout(timeOut);
    await driver.get(helper.site_url());
    await helper.login(driver, "agileway", "testwise");
  });

  after(function() {
    if (!helper.is_debugging()) {
      driver.quit();
    }
  });

  it('[4] Can enter passenger details', async function() {
    this.timeout(timeOut);

    let flight_page = new FlightPage(driver);
    await flight_page.selectTripType("oneway")
    await flight_page.selectDepartFrom("New York")
    await flight_page.selectArriveAt("Sydney")
    await flight_page.selectDepartDay("02")
    await flight_page.selectDepartMonth("May 2016")
    await flight_page.clickContinue()

    let passenger_page = new PassengerPage(driver)
    await passenger_page.clickNext();
    await driver.findElement(By.tagName("body")).getText().then(function(the_page_text) {
      assert(the_page_text.includes("Must provide last name"))
    });

    await passenger_page.enterFirstName("Bob")
    await passenger_page.enterLastName("Tester")
    await passenger_page.clickNext();

    // purposely an assertion failure
    await driver.findElement(By.name("holder_name")).getAttribute("value").then(function(val) {
      assert.equal("Wendy Tester", val)
    });

  });


});

File: selenium-webdriver-rspec/spec/03_passenger_spec.rb
load File.dirname(__FILE__) + '/../test_helper.rb'

describe "Passenger" do
  include TestHelper

  before(:all) do
    # by setting $driver global variable for build_rspec_formatter can save screenshot if error occurs
    @driver = $driver = Selenium::WebDriver.for(browser_type, browser_options)
    driver.manage().window().resize_to(1280, 720)
    driver.manage().window().move_to(30, 78)
    driver.get(site_url)
    
    login_page = LoginPage.new(driver)
    login_page.login("agileway", "testwise")
  end

  after(:all) do
    fail_safe { driver.find_element(:link_text, "Sign off").click } unless debugging?
    driver.quit unless debugging?
  end

  it "[4] Can enter passenger details (using page objects)" do
    flight_page = FlightPage.new(driver)
    flight_page.select_trip_type("return")
    flight_page.select_depart_from("Sydney")
    flight_page.select_arrive_at("New York")

    flight_page.select_depart_day("02")
    flight_page.select_depart_month("May 2016")
    flight_page.select_return_day("04")
    flight_page.select_return_month("June 2016")
    flight_page.click_continue

    # now on passenger page
    passenger_page = PassengerPage.new(driver)
    passenger_page.click_next
    try_for(3) { expect(page_text).to include("Must provide last name") }
    passenger_page.enter_first_name("Bob")
    passenger_page.enter_last_name("Tester")
    passenger_page.click_next

    expect(driver.find_element(:name, "holder_name").attribute("value")).to eq("Wendy Tester")
  end

end
